apis/v1/gateway_types.go

```diff
@@ -315,6 +315,7 @@ type ListenerNamespaces struct {
 	//
 	// +optional
 	// +kubebuilder:default=None
+	// +kubebuilder:validation:Enum=All;Selector;Same;None
```

Comment:
FWIW I still don't think we need `None` since it can be represented as not setting `allowedListeners`. We don't really need 2 ways to represent the same thing?

Note we do not have a default at the GatewaySpec level, only at the AllowedListeners level